pkgs/build-support/node/fetch-yarn-deps/fixup.js

```diff
@@ -12,6 +12,15 @@ const fixupYarnLock = async (lockContents, verbose) => {
 	const fixedData = Object.fromEntries(
 		Object.entries(lockData.object)
 		.map(([dep, pkg]) => {
+			if (pkg.resolved === undefined) {
+				console.warn(`no resolved URL for package ${dep}`)
+				var maybeFile = dep.split("@", 2)[1]
+				if (maybeFile.startsWith("file:")) {
+					console.log(`Rewriting URL for local file dependency ${dep}`)
+					pkg.resolved = maybeFile
+				}
+				return [dep, pkg]
+			}
```

Comment:
I'd really appreciate if someone would be able to comment on this change... Other then that this should be good to go.

Reply:
I can only confirm that, when building my `ente-web` package (see above) which has local file dependencies, this worked correctly. The relevant part of the output:
```
no resolved URL for package ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
Rewriting URL for local file dependency ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
no resolved URL for package photoswipe@file:./apps/photos/thirdparty/photoswipe
Rewriting URL for local file dependency photoswipe@file:./apps/photos/thirdparty/photoswipe
```
Cannot comment on the implementation itself, just that it worked for me as expected.

Reply:
> I can only confirm that, when building my `ente-web` package (see above) which has local file dependencies, this worked correctly. The relevant part of the output:
> 
> ```
> no resolved URL for package ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
> Rewriting URL for local file dependency ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
> no resolved URL for package photoswipe@file:./apps/photos/thirdparty/photoswipe
> Rewriting URL for local file dependency photoswipe@file:./apps/photos/thirdparty/photoswipe
> ```
> 
> Cannot comment on the implementation itself, just that it worked for me as expected.

Oh excellent, thanks!